### PR TITLE
DDF-04533 Migrate list-create to React

### PR DESCRIPTION
**Pull request by zta6**
#### What does this PR do?
Migrates list-create to React
Conditionally render editor-footer in list-editor based on a variable, rather than hiding via inherited css/less from list-create. 

#### Who is reviewing it? 
@Schachte 
@middlets719 
@gordocanchola 
@adimka

#### Select relevant component teams: 
@codice/ui 

#### Ask 2 committers to review/merge the PR and tag them here.
@andrewkfiedler
@bdeining

#### How should this be tested?

Open a workspace
Navigate to 'Lists'
Create a new list
Verify that the created list shows up. 
Navigate to 'Search'
Perform a search that returns some results.
On any result click the '+' icon
Click 'Create New List' and create a list
Verify that the list was successfully created and shows up in the Lists tab

#### What are the relevant tickets?
For GH Issues:
Fixes: #4533 

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: ui/packages/catalog-ui-search/src/main/webapp/component/list-create/list-create.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
+/**
```
Comment (samuelechu) [inline ui/packages/catalog-ui-search/src/main/webapp/component/list-create/list-create.js]: You'll want to separate the react part out into its own file in the directory /webapp/react-component/[presentation or container] and have the file use typescript

File: ui/packages/catalog-ui-search/src/main/webapp/component/list-create/list-create.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+      buttonText = 'Create New List with Result(s)'
+      onClickFunction = this.createList.bind(this)
+    } else {
+      buttonText = 'Create New List'
+      onClickFunction = this.createListWithBookmarks.bind(this)
+    }
+    return (
+      <CreateContainer>
+        <MarionetteRegionContainer view={this.listEditor} />
+        <div className="editor-footer">
```
Comment (adimka) [inline ui/packages/catalog-ui-search/src/main/webapp/component/list-create/list-create.js]: Is this being targeted by something? If not, we can remove the className. If a parent is styling this, we should move that styling into this file and remove the className.

File: ui/packages/catalog-ui-search/src/main/webapp/component/list-create/list-create.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+          <Button className="is-primary" onClick={() => onClickFunction()}>
```
Comment (adimka) [inline ui/packages/catalog-ui-search/src/main/webapp/component/list-create/list-create.js]: `is-primary` is a mixin from less. We should pull those styles into our Button styled component and remove this className.

File: ui/packages/catalog-ui-search/src/main/webapp/component/list-create/list-create.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+var Marionette = require('marionette')
+
+const ListCreateWithListenTo = withListenTo(ListCreate)
+
+const ListCreateView = Marionette.LayoutView.extend({
+  template() {
+    return (
+      <ListCreateWithListenTo
+        model={this.model}
```
Comment (willwill96) [inline ui/packages/catalog-ui-search/src/main/webapp/component/list-create/list-create.js]: It doesn't look like you're using this prop in your component. Can it be removed?

File: ui/packages/catalog-ui-search/src/main/webapp/component/list-create/list-create.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+ * prepare derivative works, distribute copies to the public, and perform and display publicly, in
+ * any manner and for any purpose, and to have or permit others to do so.
+ */
+import React from 'react'
+import ListCreate from '../../react-component/presentation/list-create/list-create'
+import withListenTo from 'catalog-ui-search/src/main/webapp/react-component/container/backbone-container'
+
+var Marionette = require('marionette')
+
+const ListCreateWithListenTo = withListenTo(ListCreate)
```
Comment (willwill96) [inline ui/packages/catalog-ui-search/src/main/webapp/component/list-create/list-create.js]: ✏️ I think convention has been to export components with `withListenTo` if they need it.

File: ui/packages/catalog-ui-search/src/main/webapp/component/list-create/list-create.js (hunk 4 above)
Comment (zta6) [reply ui/packages/catalog-ui-search/src/main/webapp/component/list-create/list-create.js]: It gets used here: https://github.com/codice/ddf/pull/4536/files/a951f1e141f4c8edace811e309c7e505a1290c29#diff-a85b0f2201e11c353e46dd9aeeb5bfdcR50

**Comment by gordocanchola** [comment]
build now

**Comment by gordocanchola** [comment]
Root level quick build and full build of changes were successful.
Manual verification was successful.

**Pull request merged**